reVX/handlers/sc_points.py

```diff
@@ -0,0 +1,618 @@
+
+    @property
+    def resource_gids(self):
+        """
+        Resource gids associated with supply curve point
+
+        Returns
+        -------
+        ndarray
+        """
+        return self._res_gids
+
+    @property
+    def gid_counts(self):
+        """
+        Resource gid exclusion pixel counts
+
+        Returns
+        -------
+        ndarray
+        """
+        return self._gid_counts
+
+    @property
+    def resource_capacity(self):
+        """
+        Capacity of resource gids
+
+        Returns
+        -------
+        ndarray
+        """
+        return self._res_capacity
+
+    @property
+    def cf_means(self):
+        """
+        Capacity factor means for each resource gid
+
+        Returns
+        -------
+        ndarray
+        """
+        return self._cf_means
+
+    @staticmethod
+    def _parse_list(list_in, dtype=int):
+        """
+        Parse json list if needed
+
+        Parameters
+        ----------
+        list_in : str | list
+            List or jsonified list from supply curve table
+        dtype : np.dtype
+            dtype for output array
+
+        Returns
+        -------
+        out : ndarary
+            vector of values
+        """
+        if isinstance(list_in, str):
+            list_in = json.loads(list_in)
+        elif not isinstance(list_in, list):
+            msg = ('Must supply either a jsonified list or a list, not {}'
+                   .format(type(list_in)))
+            logger.error(msg)
+            raise ValueError(msg)
+
+        out = np.array(list_in, dtype=dtype)
+
+        return out
+
+    def _drop(self, build_capacity, drop=None):
+        """
+        Drop capacity from Supply Curve point
+
+        Parameters
+        ----------
+        drop : int | None, optional
+            Number of gids to drop, if None drop all, by default None
+
+        Returns
+        -------
+        sc_point : pd.Series
+            Resource gids being allocated
+        capacity : float
+            Capacity being allocated
+        availability : bool
+            Whether Supply Curve point still has available capacity
+        """
+        drop_slice = slice(0, drop, None)
+        capacity = np.sum(self.resource_capacity[drop_slice])
+        if capacity < build_capacity:
+            build_capacity = capacity
+
+        sc_point = {'sc_gid': self.sc_gid,
+                    'res_gids': self.resource_gids[drop_slice].tolist(),
+                    'gid_counts': self.gid_counts[drop_slice].tolist(),
+                    'cf_means': self.cf_means[drop_slice].tolist(),
+                    'build_capacity': build_capacity}
+        sc_point = pd.Series(sc_point)
+
+        if drop is None:
+            self._res_gids = None
+            self._gid_counts = None
+            self._res_capacity = None
+            self._cf_means = None
+            self._capacity = 0.0
+        else:
+            self._res_gids = np.delete(self._res_gids, drop_slice)
```

Comment:
Why wouldnt you just set the capacity to zero for the "dropped" resource gid? I think setting available capacity to zero makes more sense than removing array indices.

Reply:
added a new list to modify called _avail_cap